File: packages/audio-server/src/providers.py
import os

def get_provider_names() -> dict[str, str]:
    return {
        "wake_word": os.getenv("WAKE_WORD_PROVIDER", "stub"),
        "stt": os.getenv("STT_PROVIDER", "stub"),
        "tts": os.getenv("TTS_PROVIDER", "stub"),
        "speaker_id": os.getenv("SPEAKER_ID_PROVIDER", "stub"),
    }

File: config/providers.ts
export const providers = {
  wakeWord: process.env.WAKE_WORD_PROVIDER ?? "open_wake_word",
  stt: process.env.STT_PROVIDER ?? "openai_whisper_api",
  tts: process.env.TTS_PROVIDER ?? "piper",
  speakerId: process.env.SPEAKER_ID_PROVIDER ?? "resemblyzer",
} as const;

export type WakeWordProviderName = typeof providers.wakeWord;
export type STTProviderName = typeof providers.stt;
export type TTSProviderName = typeof providers.tts;
export type SpeakerIDProviderName = typeof providers.speakerId;
